frontend/packages/console-shared/src/utils/validation/validation.ts

```diff
@@ -0,0 +1,102 @@
+import { addMissingSubject, joinGrammaticallyListOfItems, makeSentence } from '../grammar';
+import { ValidationErrorType, ValidationObject } from './types';
+import {
+  DNS1123_END_ERROR,
+  DNS1123_START_END_ERROR,
+  DNS1123_START_ERROR,
+  DNS1123_TOO_LONG_ERROR,
+  EMPTY_ERROR,
+} from './strings';
+
+const alphanumericRegex = '[a-zA-Z0-9]';
+
+const DNS_1123_OFFENDING_CHARACTERS = {
+  ',': 'comma',
+  "'": 'apostrophe', // eslint-disable-line quotes
+  _: 'underscore',
+};
+
+export const getValidationObject = (
```

Comment:
the name maybe a bit misleading - maybe we can come up with a better name? This is esentially a constuctor - not a getter. Maybe something like `asValidationObject() ` ?

Reply:
I don't think the name will confuse anyone, but `asValidationObject()` is a bit clearer. I'll update it. =)

Reply:
Done!